app/Models/User.php

```diff
@@ -91,6 +91,11 @@ public function shared()
 		return $this->belongsToMany('App\Models\Album', 'user_album', 'user_id', 'album_id');
 	}
 
+	public function get_username(): string
+	{
+		return $this->is_admin() ? 'Admin' : $this->username;
```

Comment:
What's going on here? I thought that the admin account could be called anything one wants?

Reply:
When you return the album, you have the owner username: `$album->owner->username`.
Now with the admin as `user(id = 0)`, the value of username is fixed to the bcrypt of the admin username (just like previously in the config table). As a result, `$album->owner->username` was leaking the admin username hash.

Reply:
Right... So here's a question: is it really important to bcrypt the admin username? Why not make it consistent for all users and store the username verbatim?

Reply:
It is not really important to be honest, it somewhat served as a second password.

It is just that applying the migration is going to be "difficult" as we would need to have code that manage the migration (once again in the LEGACY module) and we would still need to manage the case where the migration is not done yet.

Or we could just tell all our users _"screw you, please reset your admin username/password"_, but I'm not sure that would be appreciated with regard to PR. :sweat_smile: